File: ws/src/tabit/tabit/01_talker.py
from time import sleep

import rclpy
from std_msgs.msg import String


def main():
    rclpy.init()
    node = rclpy.create_node("my_node")
    publisher = node.create_publisher(String, "my_topic", 10)
    msg = String(data="hello there")

    while rclpy.ok():
        publisher.publish(msg)
        sleep(1.0)


if __name__ == "__main__":
    main()

File: ws/src/demo_js/01_talker.js
const rclnodejs = require('rclnodejs');

async function createPublisher() {
    await rclnodejs.init();
    const node = new rclnodejs.Node('my_js_node');

    // Create a publisher for String messages on 'my_topic'
    const publisher = node.createPublisher('std_msgs/msg/String', 'my_topic');

    setInterval(() => {
        const message = "hello there from JavaScript!";
        publisher.publish(message);
    }, 1000);

    node.spin();
}

createPublisher().catch(console.error);